src/ucp/dt/dt.c

```diff
@@ -8,21 +8,100 @@
 
 #include <ucp/core/ucp_ep.inl>
 #include <ucp/core/ucp_request.h>
+#include <ucp/core/ucp_mm.h>
 #include <ucs/profile/profile.h>
 
-size_t ucp_dt_pack(ucp_datatype_t datatype, void *dest, const void *src,
+UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_mem_type_unpack(ucp_worker_h worker, void *buffer, const void *recv_data,
```

Comment:
let's wrap this with UCS_PROFILE_FUNC
also, need to remove UCS_F_ALWAYS_INLINE , it's useless in c file..
same comment for ucp_mem_type_pack